src/node_sqlite.cc

```diff
@@ -239,6 +256,175 @@ void DatabaseSync::Exec(const FunctionCallbackInfo<Value>& args) {
+
+      if (table_value->IsString()) {
+        String::Utf8Value str(env->isolate(), table_value);
+        table = std::string(*str);
+      } else {
+        node::THROW_ERR_INVALID_ARG_TYPE(
+            env->isolate(), "The \"options.table\" argument must be a string.");
+        return;
+      }
+    }
+
+    Local<String> db_key =
+        String::NewFromUtf8(env->isolate(), "db", NewStringType::kNormal)
+            .ToLocalChecked();
+    if (options->HasOwnProperty(env->context(), db_key).FromJust()) {
+      Local<Value> db_value =
+          options->Get(env->context(), db_key).ToLocalChecked();
+      if (db_value->IsString()) {
+        String::Utf8Value str(env->isolate(), db_value);
+        dbName = std::string(*str);
+      } else {
+        node::THROW_ERR_INVALID_ARG_TYPE(
+            env->isolate(), "The \"options.db\" argument must be a string.");
+        return;
+      }
+    }
+  }
+
+  DatabaseSync* db;
+  ASSIGN_OR_RETURN_UNWRAP(&db, args.This());
+  THROW_AND_RETURN_ON_BAD_STATE(env, !db->IsOpen(), "database is not open");
+
+  sqlite3_session* pSession;
+  int r = sqlite3session_create(db->connection_, dbName.c_str(), &pSession);
+  CHECK_ERROR_OR_THROW(env->isolate(), db->connection_, r, SQLITE_OK, void());
+  db->sessions_.insert(pSession);
+
+  r = sqlite3session_attach(pSession, table == "" ? nullptr : table.c_str());
+  CHECK_ERROR_OR_THROW(env->isolate(), db->connection_, r, SQLITE_OK, void());
+
+  BaseObjectPtr<Session> session =
+      Session::Create(env, BaseObjectWeakPtr<DatabaseSync>(db), pSession);
+  args.GetReturnValue().Set(session->object());
+}
+
+static std::function<int()> conflictCallback;
+
+static int xConflict(void* pCtx, int eConflict, sqlite3_changeset_iter* pIter) {
+  if (!conflictCallback) return SQLITE_CHANGESET_ABORT;
+  return conflictCallback();
+}
+
+static std::function<bool(std::string)> filterCallback;
+
+static int xFilter(void* pCtx, const char* zTab) {
+  if (!filterCallback) return 1;
+
+  return filterCallback(zTab) ? 1 : 0;
+}
+
+void DatabaseSync::ApplyChangeset(const FunctionCallbackInfo<Value>& args) {
+  conflictCallback = nullptr;
+  filterCallback = nullptr;
+
+  DatabaseSync* db;
+  ASSIGN_OR_RETURN_UNWRAP(&db, args.This());
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_ON_BAD_STATE(env, !db->IsOpen(), "database is not open");
+
+  if (!args[0]->IsUint8Array()) {
+    node::THROW_ERR_INVALID_ARG_TYPE(
+        env->isolate(), "The \"changeset\" argument must be a Uint8Array.");
+    return;
+  }
+
+  if (args.Length() > 1 && !args[1]->IsUndefined()) {
+    if (!args[1]->IsObject()) {
+      node::THROW_ERR_INVALID_ARG_TYPE(
+          env->isolate(), "The \"options\" argument must be an object.");
+      return;
+    }
+
+    Local<Object> options = args[1].As<Object>();
+
+    Local<String> conflictKey =
+        String::NewFromUtf8(
+            env->isolate(), "onConflict", NewStringType::kNormal)
+            .ToLocalChecked();
```

Comment:
Likely better to add this to the set of strings available via `Environment`... e.g. `env->onconflict_string()`

Reply:
Could you give me a hint how to do that?

Reply:
Take a look at env_properties.h